storage/innobase/trx/trx0roll.cc

```diff
@@ -776,13 +779,18 @@ void trx_rollback_recovered(bool all)
         before trx_sys.deregister_rw(trx). */
         trx_sys.deregister_rw(trx);
         trx_free_at_shutdown(trx);
-      }
-      else
+      } else {
         trx->free();
+        DBUG_EXECUTE_IF("print_log_after_rollback_on_recovery", {
+          ib::info() << "One trx rollback by recovery_rollback_thread "
+                     << current_thd << "\n";
+          });
+      }
```

Comment:
I do not see any test coverage of this. I am not a friend of such added printouts; such information can be better collected with `./mtr --rr` and checked in `rr replay`. If this is really needed, then I would suggest `DBUG_PRINT`, which should be disabled by `cmake -DWITH_DBUG_TRACE=OFF`.

Reply:
It is used as restart-parameter when restarting the server, then rollback thread will log a message for every finished trx. the test will check existing of the log.